azurerm/internal/services/compute/linux_virtual_machine_resource.go

```diff
@@ -0,0 +1,979 @@
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			// Required
+			"admin_username": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"network_interface_ids": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: networkValidate.NetworkInterfaceID,
+				},
+			},
+
+			"os_disk": virtualMachineOSDiskSchema(),
+
+			"size": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			// Optional
+			"additional_capabilities": virtualMachineAdditionalCapabilitiesSchema(),
+
+			"admin_password": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				Sensitive:        true,
+				DiffSuppressFunc: adminPasswordDiffSuppressFunc,
+			},
+
+			"admin_ssh_key": SSHKeysSchema(true),
+
+			"allow_extension_operations": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  true,
+			},
+
+			"availability_set_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: computeValidate.AvailabilitySetID,
+				// the Compute/VM API is broken and returns the Resource Group name in UPPERCASE :shrug:
+				DiffSuppressFunc: suppress.CaseDifference,
+				// TODO: raise a GH issue for the broken API
+				// availability_set_id:                 "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/acctestRG-200122113424880096/providers/Microsoft.Compute/availabilitySets/ACCTESTAVSET-200122113424880096" => "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/acctestRG-200122113424880096/providers/Microsoft.Compute/availabilitySets/acctestavset-200122113424880096" (forces new resource)
+				ConflictsWith: []string{
+					// TODO: "virtual_machine_scale_set_id"
+					"zone",
+				},
+			},
+
+			"boot_diagnostics": bootDiagnosticsSchema(),
+
+			"computer_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+
+				// Computed since we reuse the VM name if one's not specified
+				Computed: true,
+				ForceNew: true,
+				// note: whilst the portal says 1-15 characters it seems to mirror the rules for the vm name
+				// (e.g. 1-15 for Windows, 1-63 for Linux)
+				ValidateFunc: ValidateLinuxName,
+			},
+
+			"custom_data": base64.OptionalSchema(true),
+
+			"dedicated_host_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true, // TODO: investigate, looks like the Portal allows migration
+				ValidateFunc: computeValidate.DedicatedHostID,
+				// the Compute/VM API is broken and returns the Resource Group name in UPPERCASE :shrug:
+				DiffSuppressFunc: suppress.CaseDifference,
+				// TODO: raise a GH issue for the broken API
```

Comment:
as above - will put this together once the Beta's live since it's not blocking / I want to get a full repro down so this doesn't get blamed as "an issue with the Go SDK" again 🙃